src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -2,40 +2,91 @@ package dpla.ingestion3.mappers.providers
+import org.apache.commons.codec.digest.DigestUtils
 
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
 /**
   * Interface that all provider extractors implement.
   */
 
 trait Extractor {
   // Base item uri
-  private val baseItemUri = "http://dp.la/api/items/"
+  private val baseDplaItemUri = "http://dp.la/api/items/"
 
   def build(): Try[OreAggregation]
   def agent: EdmAgent
+
+  /**
+    * Does the provider use a prefix (typically their provider abbreviation) to
+    * salt the identifier?
+    *
+    * @return Boolean
+    */
+  def useProviderName(): Boolean
+
+  /**
+    * Extract the record's "persistent" identifier. Implementations should raise
+    * an Exception if no ID can be extracted
+    *
+    * @return String Record identifier
+    * @throws Exception If ID can not be extracted
+    */
+  @throws(classOf[ExtractorException])
+  def getProviderId(): String
+
   /**
-    * Build the base ID to be hashed. Implemented per provider
+    * The value to salt the identifier with. Not all providers
+    * use this. If not, an empty string should be used
+    * @return
+    */
+  def getProviderName(): String
+
+  /**
+    * Builds the ID to be hashed by either concatenating the provider's
+    * abbreviated name and the persistent identifier for the record with
+    * a double dash `--` or only the persistent identifier.
+    *
+    * Some providers use a name prefix and some do not. Please see Hub
+    * mapping documentation and individual Extractor classes.
     *
     * @return String
     */
-  def getProviderBaseId(): Option[String]
+  def buildProviderBaseId(): String = {
+    def idErrorMsg(): String = {
+      s"Unable to mint ID given values of:\n" +
+        s"useProviderName: ${useProviderName()}\n" +
+        s"getProviderName: ${getProviderName()}\n" +
+        s"getProviderId: ${getProviderId()}\n"
+    }
+
+    (useProviderName(), getProviderName().isEmpty) match {
+      // use prefix and prefix is set
+      case (true, false) => s"${getProviderName()}--${getProviderId()}"
+      // do not use prefix and prefix is not set
+      case (false, true) => getProviderId()
+      // use prefix but prefix is not set
+      case (true, true) => throw new Exception(idErrorMsg())
+      // do not use prefix but prefix value is set
+      case (false, false) => throw new Exception(idErrorMsg())
+      case _ => throw new Exception(idErrorMsg())
+    }
+  }
 
   /**
     * Hashes the base ID
     *
-    * @return MD5 hash of the base ID
+    * @return String MD5 hash of the base ID
     */
-  protected def mintDplaId(): String = Utils.generateMd5(getProviderBaseId())
+  protected def mintDplaId(): String = DigestUtils.md5Hex(buildProviderBaseId())
 
   /**
     * Builds the item URI
     *
     * @return URI
     */
-  protected def mintDplaItemUri(): URI = new URI(s"${baseItemUri}${mintDplaId()}")
+  protected def mintDplaItemUri(): URI = new URI(s"$baseDplaItemUri${mintDplaId()}")
```

Comment:
I'm thinking a DO NOT TOUCH UNLESS YOU REALLY KNOW WHAT YOU'RE DOING- style warning message might be appropriate for this method and the one above it

Reply:
Good call.